erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -298,12 +301,40 @@ def on_submit(self):
 			from erpnext.stock.doctype.serial_no.serial_no import update_serial_nos_after_submit
 			update_serial_nos_after_submit(self, "items")
 
+		if self.update_purchase_receipt == 1:
+			self.update_sl_valuation()
+
 		# this sequence because outstanding may get -negative
 		self.make_gl_entries()
 
 		self.update_project()
 		self.update_fixed_asset()
 
+	def update_sl_valuation(self):
+		for item in self.get("items"):
+			pr = item.purchase_receipt
+			pr_doc = frappe.get_doc("Purchase Receipt", pr)
+		
+			for pr_item in pr_doc.get("items"):
+				if item.item_code == pr_item.item_code:
```

Comment:
Instead of `item_code`, update it based on `pr_detail` value which holds the value of name of the PR Item row.